src/connector.cpp

```diff
@@ -5193,19 +5025,92 @@ void HostConnector::hostSwitchPreset(const Current& prev)
                     if (defpropdata.value == defpropdata.meta.defpath)
                         continue;
 
-                    _host.patch_set(hbp.id, defpropdata.uri.c_str(), defpropdata.value.c_str());
-
-                    if (hbp.pair != kMaxHostInstances)
-                        _host.patch_set(hbp.pair, defpropdata.uri.c_str(), defpropdata.value.c_str());
+                    hostPatchSetBlockPair(hbp, defpropdata);
                 }
 
-                _host.params_flush(hbp.id, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params.size(), params.data());
-
-                if (hbp.pair != kMaxHostInstances)
-                    _host.params_flush(hbp.pair, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params.size(), params.data());
+                hostParamsFlushBlockPair(hbp, LV2_KXSTUDIO_PROPERTIES_RESET_FULL, params);
             }
         }
+        // ensure necessary dual mono blocks are added
+        hostEnsureStereoChain(prev.preset, 0, 0);
+    }
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+bool HostConnector::hostLoadInstance(const Block& blockdata, const uint16_t instance_number, const bool active)
+{
+    if (active ? _host.add(blockdata.uri.c_str(), instance_number)
+               : _host.preload(blockdata.uri.c_str(), instance_number))
+    {
+        if (!blockdata.enabled)
+            _host.bypass(instance_number, true);
+
+        for (const Parameter& paramdata : blockdata.parameters)
+        {
+            if (isNullURI(paramdata.symbol))
+                break;
+            if ((paramdata.meta.flags & (Lv2PortIsOutput|Lv2ParameterVirtual)) != 0)
+                continue;
+            if (isNotEqual(paramdata.value, paramdata.meta.def2))
+                _host.param_set(instance_number, paramdata.symbol.c_str(), paramdata.value);
```

Comment:
on cases where we iterate over all params, it is best to use the flush_param stuff instead as it is less messages and thus happens slightly faster.